components/common/navBarComponent/navBarcomponent.tsx

```diff
@@ -0,0 +1,188 @@
+import Image from "next/image";
+import navItem from "../../../models/nav/navItem";
+import navList from "../../../models/nav/navList";
+import { hideBasedOnScopes } from "../../../utils/front-end-util/frontendUtil";
+import { Navbar, Nav, Avatar, Stack } from "rsuite";
+import "rsuite/dist/rsuite.min.css";
+import logo from "../../../images/asgardeo-logo-transparent.png";
+import NavData from "../../../components/common/navBarComponent/data/nav.json";
+import { useRouter } from "next/router";
+import { useSession } from "next-auth/react";
+import { signout } from "../../../utils/authorization-config-util/authorizationConfigUtil";
+
+export interface NavComponentProps {
+  activeKeyNav: string | undefined;
+  activeKeyNavSelect: (event: string | undefined) => void;
+}
+
+export function NavBarComponent(prop: NavComponentProps) {
+  const { activeKeyNav, activeKeyNavSelect } = prop;
+  const { data: session, status } = useSession();
+  const NavConfigList: navList = NavData;
+  const router = useRouter();
+
+  const signOutOnClick = (
+    eventKey: string,
+    event: React.SyntheticEvent
+  ): void => {
+    signout(session!);
+  };
+
+  const handleNavItemSelect = (
+    eventKey: string,
+    event: React.SyntheticEvent
+  ) => {
+    // Find the selected item based on the eventKey
+    const selectedItem = findItemByEventKey(NavConfigList.items, eventKey);
+    if (selectedItem) {
+      // Construct the full route and navigate
+      const fullRoute = getFullRoute(selectedItem, session);
+      router.push(fullRoute);
+    }
+  };
+
+  // Function to find an item by eventKey in the items array
```

Comment:
Use doc comment notation to document functions.

Reply:
Fixed